hydroflow/Cargo.toml

```diff
@@ -63,18 +63,29 @@ serde = { version = "1", features = [ "derive" ] }
 serde_json = "1"
 slotmap = "1.0"
 static_assertions = "1.1.0"
-tokio = { version = "1.16", features = [ "full" ] }
 tokio-stream = { version = "0.1.10", features = [ "io-util" ] }
-tokio-util = { version = "0.7.4", features = [ "net", "codec" ] }
 variadics = { path = "../variadics" }
 rustc-hash = "1.1.0"
 tempfile = { version = "3.3.0", optional = true }
 
+[target.'cfg(not(target_arch = "wasm32"))'.dependencies]
+tokio = { version = "1.16", features = [ "full" ] }
+tokio-util = { version = "0.7.4", features = [ "net", "codec" ] }
+
+[target.'cfg(target_arch = "wasm32")'.dependencies]
+tokio = { version = "1.16", features = [ "rt" , "sync", "macros", "io-util", "time" ] }
+tokio-util = { version = "0.7.4", features = [ "codec" ] }
+# We depend on getrandom transitively through rand. To compile getrandom to
+# WASM, we need to enable its "js" feature. However, rand does not expose a
+# passthrough to enable "js" on getrandom. As a workaround, we enable the
+# getrandom js feature here; when the feature resolver computes the set of
+# features for getrandom (unification), it will include "js".
+getrandom = { version = "0.2.6", features = [ "js" ] }
+
 [dev-dependencies]
-chrono = { version = "0.4.20", features = [ "serde" ], default-features = false }
+chrono = { version = "0.4.20", features = [ "serde", "clock" ], default-features = false }
```

Comment:
I think I have to turn on clock here manually because we now use the v2 resolver; without adding this feature hydroflow fails to compile with the error:

```
   Compiling hydroflow v0.1.0 (/Users/tylerhou/code/hydroflow/hydroflow)
error[E0599]: no function or associated item named `now` found for struct `chrono::Utc` in the current scope
  --> hydroflow/examples/echoserver/client.rs:26:75
   |
26 | ...unwrap(), ts: Utc::now(), }, server_addr) )
   |                       ^^^ function or associated item not found in `Utc`

error[E0599]: no function or associated item named `now` found for struct `chrono::Utc` in the current scope
  --> hydroflow/examples/echoserver/client.rs:23:97
   |
23 | ...} from {:?}", Utc::now(), m, a));
   |                       ^^^ function or associated item not found in `Utc`
...
```

Reply:
Seems like previously we were turning on chrono transitively through a build dependency (`rust-sitter-cli`) of hydroflow_datalog_core: https://gist.github.com/tylerhou/45ef7e7ba83cef9272e74c431b244ef4

With the v2 resolver I assume this edge is treated differently (as a build-time feature) and so it's not added to the set of features for the "runtime" features of chrono.

Reply:
Yeah better this way